src/MLJInterface.jl

```diff
@@ -402,4 +402,363 @@ MLJModelInterface.metadata_model(
+    - "gamma": Gamma regression with log-link.
+    - "tweedie": Tweedie regression with log-link.
+- `categorical_feature::Vector{Int} = Vector{Int}()`: Used to specify the
+  categorical features. Items in the vector are column indices representing which
+  features should be interpreted as categorical.
+- `data_random_seed::Int = 1`: Random seed used when constructing histogram bins.
+- `is_sparse::Bool = true`: Enable/disable sparse optimization.
+- `is_unbalance::Bool = false`: Set to true if training data is unbalanced.
+- `boost_from_average::Bool = true`: Adjusts the initial score to the mean of
+  labels for faster convergence.
+- `use_missing::Bool = true`: Whether or not to handle missing values.
+- `feature_pre_filter::Bool = true`: Whether or not to ignore unsplittable
+  features.
+- `alpha::Float64 = 0.9`: Parameter used for huber and quantile regression.
+- `metric::Vector{String} = ["l2"]`: Metric(s) to be used when evaluating on
+  evaluation set. For detailed information, see [the official
+  documentation](https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric-parameters)
+- `metric_freq::Int = 1`: The frequency to run metric evaluation at.
+- `is_training_metric::Bool = false`: Set to `true` to output metric result on
+  training dataset.
+- `ndcg_at::Vector{Int} = Vector{Int}([1, 2, 3, 4, 5])`: Evaluation positions for
+  ndcg and map metrics.
+- `num_machines::Int = 1`: Number of machines to use when doing distributed
+  learning.
+- `num_threads::Int  = 0`: Number of threads to use.
+- `local_listen_port::Int = 12400`: TCP listen port.
+- `time_out::Int = 120`: Socket timeout.
+- `machine_list_file::String = ""`: Path of files that lists the machines used for
+  distributed learning.
+- `save_binary::Bool = false`: Whether or not to save the dataset to a binary file
+- `device_type::String = "cpu"`: The type of device being used. One of `cpu` or
+  `gpu`
+- `force_col_wise::Bool = false`: Force column wise histogram building. Only
+  applicable on cpu.
+- `force_row_wise::Bool = false`: Force row wise histogram building. Only
+  applicable on cpu.
+- `truncate_booster::Bool = true`: Whether or not to truncate the booster.
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `fitresult`: Fitted model information, contains a `LGBMRegression` object, an
+  empty vector, and the regressor with all its parameters
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `training_metrics`: A dictionary containing all training metrics.
+- `importance`: A `namedtuple` containing:
+    - `gain`: The total gain of each split used by the model
+    - `split`: The number of times each feature is used by the model.
+
+# Examples
+
+```julia
+
+using DataFrames: DataFrame using MLJ
+
+
+LGBMRegressor = @load LGBMRegressor
+
+X, y = @load_boston # a table and a vector X = DataFrame(X) train, test =
+partition(collect(eachindex(y)), 0.70, shuffle=true)
```

Comment:
Missing carriage return:

```suggestion
X, y = @load_boston # a table and a vector 
X = DataFrame(X) train, test = partition(collect(eachindex(y)), 0.70, shuffle=true)
```